StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -98,9 +98,18 @@ public class EmbeddedPaymentElement {
             appearance: configuration.appearance,
             shouldShowApplePay: shouldShowApplePay,
             shouldShowLink: shouldShowLink,
-            savedPaymentMethodAccessoryType: savedPaymentMethodAccessoryType
+            savedPaymentMethodAccessoryType: savedPaymentMethodAccessoryType,
+            mandateProvider: MandateProvider(configuration: configuration,
+                                             elementsSession: loadResult.elementsSession,
+                                             intent: .deferredIntent(intentConfig: intentConfiguration))
         )
-        return .init(view: embeddedPaymentMethodsView, configuration: configuration)
+
+        let embeddedPaymentElement: EmbeddedPaymentElement = .init(view: embeddedPaymentMethodsView, configuration: configuration)
+        await MainActor.run {
```

Comment:
Maybe this public API should be: `@MainActor public static func create(...`

Reply:
I think once we're interacting with UIKit it should be on the main thread. Not sure if the whole func should, wdyt?  

AFAIU we still need to use `await` no matter what when we call async methods